cmd/nsm-init/nsm-init.go

```diff
@@ -192,22 +195,57 @@ func main() {
 	}
 
 	ctx, cancel := context.WithTimeout(context.Background(), clientConnectionTimeout)
+	defer cancel()
 	conn, err := dial(ctx, clientSocket)
 	if err != nil {
 		logrus.Errorf("nsm client: failure to communicate with the socket %s with error: %+v", clientSocket, err)
 		os.Exit(1)
 	}
-	nsmClient := nsmconnect.NewClientConnectionClient(conn)
 	defer conn.Close()
-	defer cancel()
 	logrus.Infof("nsm client: connection to nsm server on socket: %s succeeded.", clientSocket)
-	logrus.Infof("nsm client: client api %+v", nsmClient)
+
 	// Init related activities start here
+	nsmClient := nsmconnect.NewClientConnectionClient(conn)
+	// Getting list of available NetworkServices on local NSM server
+
+	availablaNetworkServices, err := getNetworkServices(nsmClient)
+	if err != nil {
+		logrus.Fatalf("nsm client: failed to get a list of NetworkServices from NSM, exiting...", err)
+		os.Exit(1)
+	}
 
+	if len(availablaNetworkServices) == 0 {
+		// Since local NSM has no any NetworkServices, then there is nothing to configure for the client
+		logrus.Info("nsm client: Local NSM does not have any NetworkServices, exiting...")
+		os.Exit(0)
```

Comment:
We'd want the client to continue running so when NetworkServices are configured it would pick them up.

Reply:
By nature, init container, if we are talking about init container, if one shot deal. It does not run at the same time with main application, it runs only once, right before the main app container and then it goes away. 
I think it is a good model, because even if it were running as a side car container and had detected that NSM got new NetworkService, it would still need to restart the whole POD so the application container would start using.

Reply:
I think we may eventually want to evolve into something when we can dynamically modify the networking, but this makes sense for now. Thanks.